src/layout.jl

```diff
@@ -635,36 +630,67 @@ PlaceHolder(ren::AbstractRenderable) = PlaceHolder(ren.measure.w, ren.measure.h)
 #                                     GRID                                     #
 # ---------------------------------------------------------------------------- #
 
+"""
+    calc_nrows_ncols(n, aspect=(16, 9))
+
+Computes the number of rows and columns to fit a given number `n` of subplots in a figure with aspect `aspect`.
+Adapted from: stackoverflow.com/a/43366784
+"""
+function calc_nrows_ncols(n, aspect=(16, 9))
+    w, h = aspect
+    factor = √(n / (w * h * 1.0))
+    cols = floor(Int, w * factor)
+    rows = floor(Int, h * factor)
+    rowFirst = w < h
+    while rows * cols < n
+        if rowFirst
+            rows += 1
+        else
+            cols += 1
+        end
+        rowFirst = !rowFirst
+    end
+    return rows, cols
+end
+
 """
     grid(rens::AbstractVector{<:AbstractRenderable}, layout::Union{Nothing,Tuple} = nothing, placeholder = nothing)
 
-Construct a square grid from a `AbsractVector` of `AbstractRenderable`
+Construct a square grid from a `AbsractVector` of `AbstractRenderable`.
 """
 function grid(
     rens::AbstractVector{<:AbstractRenderable};
     layout::Union{Nothing,Tuple} = nothing, 
-    placeholder = nothing
+    placeholder = nothing,
+    aspect = (16, 9),
+    hspacer = "",
+    vspacer = hspacer,
 )
     num = length(rens)
-    if layout === nothing
-        m = ceil(Int, √(num))
+    m, n = layout === nothing ? calc_nrows_ncols(num, aspect) : layout
+    if m === nothing
+        m, r = divrem(num, n)
+        r == 0 || (m += 1)
+    elseif n === nothing
         n, r = divrem(num, m)
         r == 0 || (n += 1)
-    else
-        m, n = layout
     end
-    fill_in = placeholder === nothing ? eltype(rens)() : placeholder
-    return grid(reshape(append!(copy(rens), [fill_in for _ in 1:(m * n - num)]), m, n))
+    fill_in = placeholder === nothing ? PlaceHolder(first(rens)) : placeholder
+    return grid(
+        reshape(append!(copy(rens), [fill_in for _ in 1:(m * n - num)]), m, n);
+        hspacer = hspacer,
+        vspacer = vspacer,
+    )
 end
 
 """
     grid(rens::AbstractMatrix{<:AbstractRenderable})
 
-Construct a grid from a `AbstractMatrix` of `AbstractRenderable`
+Construct a grid from a `AbstractMatrix` of `AbstractRenderable`.
 """
-function grid(rens::AbstractMatrix{<:AbstractRenderable})
-    rows = collect(foldl(*, col[2:end], init = first(col)) for col in eachrow(rens))
-    return foldl(/, rows[2:end], init = first(rows))
+function grid(rens::AbstractMatrix{<:AbstractRenderable}; hspacer = "", vspacer = hspacer)
+    rows = collect(foldl((a, b) -> a * hspacer * b, col[2:end], init = first(col)) for col in eachrow(rens))
+    return foldl((a, b) -> a * vspacer * b, rows[2:end], init = first(rows))
```

Comment:
If I understand correctly here is where you're stacking single rows of the grid, shouldn't it be `/` instead of `*`.

If you have multiple rows, each of which is a `Renderable` you can use `vstack(rows...)` to vertically stack all of them at once, but I guess you;d have to manually insert the `vspacer` in there. 


One alternative option whould be to create the columns first and then horizontally stack them to get the grid. The advantage with this is that you could use `cvstack` to ensure renderables in each column are centered. It might help with the layout for renderables of unequal sizes?

Reply:
> If I understand correctly here is where you're stacking single rows of the grid, shouldn't it be / instead of *.

Copy-paste mistake, fixed.